plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/transport/http/OpenLineageHttpTransport.java

```diff
@@ -36,14 +36,12 @@ public class OpenLineageHttpTransport
     private final Map<String, String> urlParams;
     private final Map<String, String> headers;
 
-    private static class ApiKeyTokenProvider
+    private record ApiKeyTokenProvider(String token)
```

Comment:
Can we remove this record and use `io.openlineage.client.transports.ApiKeyTokenProvider` instead?

Reply:
`ApiKeyTokenProvider` from OpenLineage library provides a setter method for the token instead of constructor.

Reply:
I think it is better in this way - instead of using `ApiKeyTokenProvider` from OL libraray. WDYT @ebyhr ?